### PR TITLE
Wait for async processes to configure the `buildScan` extension

### PR DESCRIPTION
**Pull request by bamboo**
In `buildScan.buildFinished` to guarantee everything has finished before
transmitting the event stream.
 
- [x] Verify test coverage and [CI build status](https://builds.gradle.org/viewQueued.html?itemId=12558087)

### CODE REVIEW

File: buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt [hunk 1]
```diff
@@ -102,47 +101,46 @@ open class BuildScanPlugin : Plugin<Project> {
...
-            val latch = CountDownLatch(1)
-            thread(start = true) {
-                try {
-                    action()
-                } catch (e: Exception) {
-                    rootProject.logger.warn("Build scan user data async exec failed", e)
-                } finally {
-                    latch.countDown()
-                }
+        fun async(action: () -> Unit) = thread {
```
Comment (ldaley) [inline buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt]: This method is oddly named. Perhaps `fork` would be better.

**Comment by ldaley** [comment]
1 minor comment that is not blocking merge.

👍

**Comment by bamboo** [comment]
Done. Thanks, Luke.

**Comment by bamboo** [comment]
New feedback build: https://builds.gradle.org/viewQueued.html?itemId=12610537

**Pull request merged**